Cookbook-ios/Carousel/CarouselFlowLayout.swift

```diff
@@ -0,0 +1,134 @@
+    }
+    
+    override public func targetContentOffset(forProposedContentOffset proposedContentOffset: CGPoint, withScrollingVelocity velocity: CGPoint) -> CGPoint {
+        guard let collectionView = collectionView, !collectionView.isPagingEnabled,
+            let layoutAttributes = self.layoutAttributesForElements(in: collectionView.bounds)
+            else { return super.targetContentOffset(forProposedContentOffset: proposedContentOffset) }
+        
+        let isHorizontal = self.scrollDirection == .horizontal
+        
+        let midSide = (isHorizontal ? collectionView.bounds.size.width : collectionView.bounds.size.height) / 2
+        let proposedContentOffsetCenterOrigin = (isHorizontal ? proposedContentOffset.x : proposedContentOffset.y) + midSide
+        
+        var targetContentOffset: CGPoint
+        if isHorizontal {
+            let closest = layoutAttributes.sorted { abs($0.center.x - proposedContentOffsetCenterOrigin) < abs($1.center.x - proposedContentOffsetCenterOrigin) }.first ?? UICollectionViewLayoutAttributes()
+            targetContentOffset = CGPoint(x: floor(closest.center.x - midSide), y: proposedContentOffset.y)
+        }
+        else {
+            let closest = layoutAttributes.sorted { abs($0.center.y - proposedContentOffsetCenterOrigin) < abs($1.center.y - proposedContentOffsetCenterOrigin) }.first ?? UICollectionViewLayoutAttributes()
+            targetContentOffset = CGPoint(x: proposedContentOffset.x, y: floor(closest.center.y - midSide))
+        }
+        
+        return targetContentOffset
+    }
+}
+
+// MARK: - Helper methods
+private extension CarouselFlowLayout {
+    func setupCollectionViewDecelerationRate() {
+        guard let collectionView = collectionView else { return }
+        collectionView.decelerationRate = UIScrollView.DecelerationRate.fast
+    }
+    
+    func updateLayout() {
+        guard let collectionView = collectionView else { return }
+        let collectionSize = collectionView.bounds.size
+        let isHorizontal = (scrollDirection == .horizontal)
+        
+        let yInset = (collectionSize.height - itemSize.height) / 2
+        let xInset = (collectionSize.width - itemSize.width) / 2
+        sectionInset = UIEdgeInsets(top: yInset, left: xInset, bottom: yInset, right: xInset)
+        
+        let side = isHorizontal ? itemSize.width : itemSize.height
+        let scaledItemOffset =  (side - side * sideItemScale) / 2
+        minimumLineSpacing = sideSpacing - scaledItemOffset
+    }
+    
+    func transformLayoutAttributes(attributes: UICollectionViewLayoutAttributes) -> UICollectionViewLayoutAttributes {
+        guard let collectionView = self.collectionView else { return attributes }
+        let isHorizontal = self.scrollDirection == .horizontal
+        
+        let collectionCenter = isHorizontal ? collectionView.frame.size.width/2 : collectionView.frame.size.height/2
```

Comment:
Always leave spaces between operators (or don't, but follow the same rule in every part of the code 😉).